### ISSUE TITLE
`clearAll` method may cause all `setXXX:forKey:` methods to fail 

### ISSUE THREAD

**Issue by dongxinb** [Tencent/MMKV#359]
### The language of MMKV
Objective-C
### The version of MMKV
All version
### The platform of MMKV
iOS
### The installation of MMKV
Cocoapods
### What's the issue?

We find that if we call `-[MMKV clearAll]` method right after calling `-[MMKV clearMemoryCache]`, all `-[MMKV setXXX:forKey:]` methods will return NO.

Here is the procedure to reproduce the bug.
```objective-c
@property(nonatomic, strong) MMKV *myMMKV;

- (void)initMMKV {
    self.myMMKV = [MMKV mmkvWithID:@"test"];
}

- (void)test {
    [self.myMMKV clearMemoryCache];  // or may be triggered by Memory Warning
    [self.myMMKV clearAll];
    NSAssert([self.myMMKV setString:@"value" forKey:@"key"] == NO, @"Fail to save");
}
```

I have checked the code and find the reason:

1. `m_needLoadFromFile` will be set to `YES` after calling `-[MMKV clearMemoryCache]`.
2. In the `-[MMKV clearAll]` method, the MMKV file will be removed from file system since `m_needLoadFromFile` is true.
3. In the following instructions, there is no logic to check and create the MMKV file. Therefore, `open(m_crcPath.UTF8String, O_RDWR, S_IRWXU);` will definitely fail since the file does not exist.

Any ideas to fix the bug? Maybe we can fix it by checking and creating the MMKV file in the `-[MMKV loadFromFile]` method?

**Comment by lingol** [comment]
Noted. Will fix soon.

**Comment by lingol** [comment]
Fixed with [743d6bc](https://github.com/Tencent/MMKV/commit/743d6bc4cf1737fd6efc9040a5ffb075f484140b) on the `dev` branch.

```
	m_fd = open(m_path.UTF8String, O_RDWR | O_CREAT, S_IRWXU);
```

**Comment by dongxinb** [comment]
Thank you!

**Comment by lingol** [comment]
Released with [v1.0.24](https://github.com/Tencent/MMKV/releases/tag/v1.0.24).
